pkg/injector/webhook.go

```diff
@@ -128,9 +133,54 @@ func (wh *webhook) healthHandler(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// Helper to parse timeout variable from webhook URL
+func readTimeout(req *http.Request) (*time.Duration, error) {
+	durationValue, found := req.URL.Query()[webhookTimeoutStr]
+	if !found || len(durationValue) != 1 {
+		log.Error().Msgf("Webhook timeout value not found in request")
+		return nil, errParseWebhookTimeout
+	}
+
+	val, err := time.ParseDuration(durationValue[0])
+	if err != nil {
+		log.Error().Msgf("Error parsing timeout value as duration: %v", err)
+		return nil, errParseWebhookTimeout
+	}
+	return &val, nil
+}
+
+// Time tracking function for webhook processing.
+func webhookTimeTrack(resourceName *string, start time.Time, timeout time.Duration) {
+	elapsed := time.Since(start)
+	var logEv *zerolog.Event
+	percentOfTimeout := float64(elapsed.Microseconds()) / float64(timeout.Microseconds())
+
+	if percentOfTimeout < 0.75 {
+		logEv = log.Debug()
+	} else if percentOfTimeout < 1 {
+		logEv = log.Warn()
+	} else {
+		// Error logging when going beyond timeout value to process a webhook
+		logEv = log.Error()
+	}
+	logEv.Msgf("Mutate Webhook for %s took %v to execute (%.2f of it's timeout, %v)",
+		(*resourceName), elapsed, percentOfTimeout, timeout)
+}
```

Comment:
Please move these to a new file called `profile.go`

Reply:
fair enough